emotion/app/root.tsx

```diff
@@ -50,7 +52,9 @@ const Document = withEmotionCache(
 
       // reset cache to re-apply global styles
       clientStyleData.reset();
-    }, [clientStyleData, emotionCache.sheet]);
+      // We only want to do this on mount, not when styles change
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This won't work as expected anyway since effects may run more than once (in `StrictMode` this effect will always be called twice as of React 18)

Reply:
We could use a ref to track if we've executed the re-linking, which would solve the issue of it being called twice. If we did that, we could avoid disabling the lint rule, though I would argue this is one of the rare cases when it makes sense to disable the rule.

Reply:
If you go that route it wouldn't make a difference if the effect ran once or 1000 times really, so I'd opt not to disable the rule to keep examples minimal and clean (and to prevent folks from thinking we're sneakily breaking rules when we're not).

Reply:
I'll make that change tonight, hopefully we can get this merged!

Reply:
Btw, this example still uses React 17's `hydrate` and `renderToString` if I'm not mistaken.

Reply:
We can bump to React 18, but I'm pretty sure it needs to use `renderToString` if styles are rendered in the head vs. adjacent to component elements. The docs suggest streaming APIs are not supported with the "advanced" approach. https://emotion.sh/docs/ssr#advanced-approach

Reply:
Yeah, there's also https://emotion.sh/docs/ssr#renderstylestonodestream, but I'm super confused.

I arrived here because I have two projects, one built with Next and the other with Remix, and both use MUI (and hence Emotion). I'm trying to upgrade both to React 18 and I'm having a hard time understanding everything.

Reply:
Welcome to our world! We're trying to get people out of this stuff for a reason 😅